src/endpoint/s3/s3_bucket_logging.js

```diff
@@ -0,0 +1,126 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const dbg = require('../../util/debug_module')(__filename);
+const http_utils = require('../../util/http_utils');
+
+function is_bucket_logging_enabled(source_bucket, bucket_name) {
+
+    if (!source_bucket || !source_bucket.bucket_info.logging) {
+        dbg.info("Bucket logging is not configured for bucket : ", bucket_name);
+        return false;
+    }
+
+    const enable = check_logging_config_perm(source_bucket.bucket_info.logging);
+    return enable;
+
+}
+
+function check_logging_config_perm(logging) {
+    // TODO: check if log bucket has permission to upload log objects and other things.
+    return true;
+}
+
+function endpoint_bucket_op_logs(op_name, req, res, source_bucket) {
+
+    dbg.info("Sending op logs for op name = ", op_name);
+    // 1 - Get all the information to be logged in a log message.
+    // 2 - Format it and send it to log bucket/syslog. 
+
+    const s3_log = get_bucket_log_record(op_name, source_bucket, req, res);
+    send_bucket_op_logs(s3_log);
+
+}
+
+function get_bucket_log_record(op_name, source_bucket, req, res) {
+
+    const client_ip = http_utils.parse_client_ip(req);
+    // const bucket_url = _get_redirection_bucket(req, req.params.bucket);
+
+    const log = {
+        op: req.method,
+        bucket_owner: source_bucket.bucket_owner,
+        source_bucket: req.params.bucket,
+        object_key: req.originalUrl,
+        log_bucket: source_bucket.bucket_info.logging.log_bucket,
+        remote_ip: client_ip,
+        request_uri: req.originalUrl,
+        http_status: res.statusCode,
+        request_id: req.request_id
+    };
+
+    const s3_log = new Aws_log_entry();
+
+    s3_log.bucket_owner = log.bucket_owner;
+    s3_log.bucket = log.source_bucket;
+    s3_log.remote_ip = log.remote_ip;
+    if (log.requester) {
+        s3_log.requester = log.requester;
+    }
+    s3_log.request_id = log.request_id;
+    s3_log.operation = log.op;
+    s3_log.key = log.object_key;
+    s3_log.http_status = log.http_status;
+    if (log.error_code) {
+        s3_log.error_code = log.error_code;
+    }
+    // s3_log.request_uri = log.request_uri;
+
+    return s3_log;
+}
+
+async function send_bucket_op_logs(s3_log) {
+
+    let values = "";
+    for (const key in s3_log) {
+        if (s3_log.hasOwnProperty(key)) {
+            const value = s3_log[key];
+            values += `${value} `;
+        }
+    }
+    values = values.slice(0, -2);
+    //Just to log it to endpoing logs. Need to remove it later.
+    dbg.error("!!!Object operation log!!! :", values);
+
+}
+
+function Aws_log_entry(log_entry) {
```

Comment:
eventually we do want to produce a log in the S3 format, so it can be used for log-based replication.
We might want to format it later, before storing the log object itself (in the next PRs)